index.html

```diff
@@ -5,77 +5,151 @@
 	<meta charset=utf-8>
 	<title>Wireguard Key Generator (Trustless)</title>
 	<meta name="author" content="Juan I Carrano">
-	<script src="wireguard.js"
-		integrity="sha256-KnkrBLei0lXuE9dGbMURxP000Jg7a9Tm8Gvvc6DysrU"
+	<script src="wireguard.js" integrity="sha256-KnkrBLei0lXuE9dGbMURxP000Jg7a9Tm8Gvvc6DysrU="
 		crossorigin="anonymous"></script>
 	<style>
-		body{margin:40px
-		auto;max-width:650px;line-height:1.6;font-size:18px;color:#444;padding:0
-		10px}h1,h2,h3{line-height:1.2}
-		h2 {font-size:large;}
-		div {padding: 1ex; margin-top: 1ex;}
-		div.adm {background-color: rgb(255, 231, 231);}
-		div.usr {text-align: center;}
-		#gen {background-color: rgb(248, 206, 69);}
-		#results {display: none; background-color: rgb(91, 241, 136);}
+		body {
+			margin: 40px auto;
+			max-width: 650px;
+			line-height: 1.6;
+			font-size: 18px;
+			color: #444;
+			padding: 10px
+		}
+
+		h1,
+		h2,
+		h3 {
+			line-height: 1.2
+		}
+
+		h2 {
+			font-size: large;
+		}
+
+		div {
+			padding: 1ex;
+			margin-top: 1ex;
+		}
+
+		div.adm {
+			background-color: rgb(255, 231, 231);
+		}
+
+		div.usr {
+			text-align: center;
+		}
+
+		div.usr a {
+			text-decoration: none;
+		}
+
+		#gen {
+			background-color: rgb(248, 206, 69);
+		}
+
+		.disabled {
+			filter: saturate(0);
+			opacity: 0.5;
+		}
+
+		#results {
+			display: none;
+			background-color: rgb(91, 241, 136);
+		}
+
+		pre {
+			text-align: left;
+			font-size: small;
+			background-color: rgb(230, 230, 230);
+			display: inline-block;
+			padding: 5px
+		}
 	</style>
 </head>
 
 <body>
 	<h1>Wireguard Key Generator (Trustless)</h1>
-	<div class="usr"><h2><em>Use this webpage on a private window!!!</em></h2></div>
+	<div class="usr">
+		<h2><em>Use this webpage on a private window!!!</em></h2>
+	</div>
 	<div class="adm">
-	<details>
-		<summary>Parameters</summary>
-		<form id="params" method="get">
-			<label for="sa">Server Address:</label><input type="text" id="sa" name="sa" value="example.com">
-			<label for="sp">Port:</label><input type="number" id="sp" , name="sp" min="1" max="65635" value="51820"><br>
-			<label for="sk">Server Pubkey:</label><input type="text" id="sk" , name="sk" value="RANDOM-EXAMPLE"><br>
-			<label for="ca">Client Address:</label><input type="text" id="ca" name="ca" value="10.0.0.2"><br>
-			<label for="aa">Allowed Subnet:</label><input type="text" id="aa" name="aa" value="10.0.0.0"><br>
-			<label for="ap">Allowed Prefix:</label><input type="number" id="ap" name="ap" min="0" max="32" value="24"><br>
-			<label for="ka">Keepalive:</label><input type="number" id="ka" name="ka" min="0" max="1000" value="25"><br>
-			<label for="cn">Client name (optional):</label><input type="text" id="cn" name="cn"><br>
-			<label for="ae">Admin email (optional):</label><input type="text" id="ae" name="ae"><br>
```

Comment:
Don't put it into the query string. Put it into the field of the email that's generated. There it's much easier, because the mail program can support you with features like auto suggestion and address book. The typing all together is the same.
To me, it only makes sense to put in a email address into a webform, if the webform's server itself sends the mail, not if only a email is generated, which I have to send myself anyway.

Reply:
But maybe one use case for having this field anyway could be to have it saved inside the query string. I'll check the logic of showing the email button has to be changed with having the admin email form back.